code/game/say.dm

```diff
@@ -111,24 +118,38 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	if(copytext_char(input, -2) == "!!")
 		spans |= SPAN_YELL
 
+	var/say_mod = message_mods[MODE_CUSTOM_SAY_EMOTE]
+	if(!say_mod)
+		say_mod = say_mod(input, message_mods)
+
 	var/spanned = attach_spans(input, spans)
-	return "[say_mod(input, message_mods)], \"[spanned]\""
+	return "[say_mod], \"[spanned]\""
+
+/// Transforms the speech emphasis mods from [/atom/movable/proc/say_emphasis] into the appropriate HTML tags. Includes escaping.
+#define ENCODE_HTML_EMPHASIS(input, char, html, varname) \
+	var/static/regex/##varname = regex("(?<!\\\\)[char](.+?)(?<!\\\\)[char]", "g");\
+	input = varname.Replace_char(input, "<[html]>$1</[html]>")
+
+/// Scans the input sentence for speech emphasis modifiers, notably |italics|, +bold+, and _underline_ -mothblocks
+/atom/proc/say_emphasis(input)
+	ENCODE_HTML_EMPHASIS(input, "\\|", "i", italics)
+	ENCODE_HTML_EMPHASIS(input, "\\+", "b", bold)
+	ENCODE_HTML_EMPHASIS(input, "_", "u", underline)
```

Comment:
I hate the + and |, bold should be * I think or the same as markdown \*\***bold**\*\*.

If the problem is that math may be picked up like "8 * 4 + 5 * 2", then we can just do what discord does and only allow it if there is no spaces between it.
```regex
\*\*[^ ](.*)[^ ]\*\*
```

I don't mind having both for people who play on multiple servers, but I think \* should either be bold, or use markdown for italics and boldness.